docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
-     suite configuration for this.
+   * Consider making all validation strict?
+     * If we really want to allow naked dummy tasks in some circumstances then
+       we would be better to add a suite configuration for this?
+     * Note: we don't allow you to inherit from an undeclared ("naked") family
+       so why do we allow you to run an undeclared task?
 
-5. Handle `rose-suite.conf` functionality
-   * Pull items from other locations / sources on installation (e.g. svn).
-   * Configure Jinja2 / EmPy inputs.
-   * Set environment variables exported to the suite daemon?
-   * Continue to handle `rose-suite.conf` using a plugin preprocessor.
-   * Decide which features to support natively in Cylc and how.
+5. Handle
+   [rose-suite.conf](https://metomi.github.io/rose/doc/html/api/configuration/suite.html#rose:file:rose-suite.conf)
+   functionality including optional configurations. See
+   [cylc-flow #3819](https://github.com/cylc/cylc-flow/issues/3819)
+   and [rose #2412](https://github.com/metomi/rose/pull/2412).
 
 6. Utility to remove installed locations (on all platforms) on demand
    (equivalent to `rose suite-clean`).
 
-7. On starting up a new run of the suite assign a new run name:
+7. Use a different run directory for each installation of a suite:
    ```
-   mi-aa001/
+   u-ab123/
      run1
      run2
    ```
-   * This is new functionality but Cylc 8 is a good time to introduce it.
+   * This is big change in working practise but Cylc 8 is a good time to
+     introduce it.
    * Use a symbolic link to identify the latest run which is the default name
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git.
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `run1` already exists it will install into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
```

Comment:
Might `run-last` be more intuitive?

Reply:
Don't really mind. I suggested `runN` because we currently use `NN` as a link to the latest job in the log directory.

Reply:
IMO `last` is somewhat ambiguous - it could mean this is the final (last ever) run, rather than just the lastest run in an ongoing series of runs.

I like `runN` because it evokes the well known (?) `1, 2, ..., N` representation of an integer series (`run1, run2, ..., runN`)

I suppose `run1, run2, ..., run-latest` might be more intuitively obvious to users, but it's a bit ugly.

Reply:
If we make the run name pattern user-definable then it might be best to use `NN` or something suitably neutral.

Reply:
Why should we make it user-definable though? (see next discussion below - surely the metadata conveyed by a less generic name for "run", which is the same for every run!, should be one level up, in the flow name?

Reply:
I suggest we leave the proposal as is - we can always come back to this idea later